lib/std/hash/rapidhash.zig

```diff
@@ -0,0 +1,128 @@
+const std = @import("std");
+
+const readInt = std.mem.readInt;
+const assert = std.debug.assert;
+const expect = std.testing.expect;
+const expectEqual = std.testing.expectEqual;
+
+const RAPID_SEED: u64 = 0xbdd89aa982704029;
+const RAPID_SECRET: [3]u64 = .{ 0x2d358dccaa6c78a5, 0x8bb84b93962eacc9, 0x4b33a62ed433d4a3 };
+
+pub const RapidHash = struct {
+    pub fn hash(seed: u64, input: []const u8) u64 {
+        return rapidhash_internal(input, input.len, seed, &RAPID_SECRET);
+    }
+};
+
+test "RapidHash.hash" {
+    const bytes: []const u8 = "abcdefgh" ** 128;
+
+    const sizes: [13]u64 = .{ 0, 1, 2, 3, 4, 8, 16, 32, 64, 128, 256, 512, 1024 };
+
+    const outcomes: [13]u64 = .{
+        0x5a6ef77074ebc84b,
+        0xc11328477bc0f5d1,
+        0x5644ac035e40d569,
+        0x347080fbf5fcd81,
+        0x56b66b8dc802bcc,
+        0xb6bf9055973aac7c,
+        0xed56d62eead1e402,
+        0xc19072d767da8ffb,
+        0x89bb40a9928a4f0d,
+        0xe0af7c5e7b6e29fd,
+        0x9a3ed35fbedfa11a,
+        0x4c684b2119ca19fb,
+        0x4b575f5bf25600d6,
+    };
+
+    var success: bool = true;
+    for (sizes, outcomes) |s, e| {
+        const r = RapidHash.hash(RAPID_SEED, bytes[0..s]);
+
+        expectEqual(e, r) catch |err| {
+            std.debug.print("Failed on {d}: {!}\n", .{ s, err });
+            success = false;
+        };
+    }
+    try expect(success);
+}
+
+inline fn mum(a: *u64, b: *u64) void {
+    const r = @as(u128, a.*) * b.*;
+    a.* = @truncate(r);
+    b.* = @truncate(r >> 64);
+}
+
+inline fn mix(a: u64, b: u64) u64 {
+    var _a = a;
```

Comment:
Please avoid prefixing identifiers with `_`, these need to be named something better.